src/syscall_mock/mod.rs

```diff
@@ -109,11 +114,15 @@ impl SyscallMock {
         }
     }
 
-    fn handle_step(&mut self, received: protocol::Command) -> R<PathBuf> {
+    fn handle_step(&mut self, received: Command) -> R<PathBuf> {
         let mock_config = match self.protocol.steps.pop_front() {
             Some(next_protocol_step) => {
-                if next_protocol_step.command != received {
-                    self.register_error(&next_protocol_step.command.format(), &received.format());
+                let expected_command = next_protocol_step.command.resolve_executable_path();
+                if expected_command != received {
```

Comment:
Maybe we should move this into a `Command.compare` function? So that the resolution of executable paths happens inside that function?

Reply:
Sounds like a good idea to me. I implemented `compare` for checking received commands against protocol steps, and `compare_executables` for checking commands against `unmocked_executables`. I was not sure whether it would be preferable stylistically to implement `compare` as `pub fn compare(&self, other: &Command)` or `pub fn compare(a: &Command, b: &Command)`. I went with the former for now.